lib/collectionspace/tools/xml.rb

```diff
@@ -491,17 +424,61 @@ def self.add_concepts(xml, field, values = [], method = :add_group_list)
           add_authorities xml, field, 'conceptauthorities', 'concept', values, method
         end
 
+        # Adds a single value structured date field group at the top level of the document
+        # arguments:
+        #  xml - the XML document
+        #  field - String. The base name of the the date field group
+        #  date - CSDTP StructuredDate. Run `CSDTP.parse(value)` on this before passing
+        #         it as a parameter
+        #  suffix - Optional, will default to 'Group'. String. Added to end of field to create
+        #           top-level field name
+        # This:
+        #  CSXML::Helpers.add_date_group(xml, 'thisDate', '1999-09-09')
+        #
+        # Will produce:
+        #
+        # <thisDateGroup>
+        #   ...structured date fields for 1999-09-09...
+        # </thisDateGroup>
+        #
+        # This:
+        #  CSXML::Helpers.add_date_group(xml, 'thisDate', '1999-09-09', '')
+        #
+        # Will produce:
+        #
+        # <thisDate>
+        #   ...structured date fields for 1999-09-09...
+        # </thisDate>
         def self.add_date_group(xml, field, date, suffix = 'Group')
           return unless date.display_date
-
           CSXML.add_group(xml, field, CSDTP.fields_for(date), suffix)
         end
 
+        # arguments:
+        #  xml - the XML document
+        #  field - String. The base name of the date field group list to be created
+        #  dates - String. Raw values from the incoming data
+        #
+        # This:
+        #  CSXML::Helpers.add_date_group_list(xml, 'thisDate', '1999-09-09; 2010-12-22')
+        #
+        # Will produce:
+        #
+        # <thisDateGroupList>
+        #  <thisDateGroup>
+        #   ...structured date fields for 1999-09-09...
+        #  </thisDateGroup>
+        #  <thisDateGroup>
+        #   ...structured date fields for 2010-12-22...
+        #  </thisDateGroup>
+        # </thisDateGroupList>
         def self.add_date_group_list(xml, field, dates)
-          dates = dates.map { |d| CSDTP.fields_for(d) }.compact
+          dates = CSDH.split_value(dates)
+          dates = dates.map{ |d| CSDTP.parse(d) }.compact
+          dates = dates.map{ |d| CSDTP.fields_for(d) }
```

Comment:
@mark-cooper @Paulmulonzia The point of a date_group_list is handling multiple date values, but this was not doing that. I think it was written to supply the field structure only. Now it handles the field structure AND repeated values within that structure.

Reply:
working perfectly. Thank you.